Formula/s/sevenzip.rb

```diff
@@ -8,7 +8,7 @@ class Sevenzip < Formula
 
   livecheck do
     url "https://7-zip.org/download.html"
-    regex(/>\s*Download\s+7-Zip\s+v?(\d+(?:\.\d+)+)(?!\s+\(beta\))[\s<]/im)
+    regex(/>\s*Download\s+7-Zip\s+v?(\d+(?:\.\d+)+)(?!\s+\(beta\))\s+\(/im)
```

Comment:
```suggestion
    regex(%r{>\s*Download\s+7-Zip\s+v?(\d+(?:\.\d+)+)\s+\([^)]+?\)(?:</?[^>]+?>)*:}im)
```

If we make this regex stricter about matching the "Download 7-Zip 23.01 (2023-06-20):" format, we can also exclude the Windows releases (e.g., "Download 7-Zip 19.00 (2019-02-21) for Windows:"). This change only returns 23.01 and omits the 19.00, 16.04, and 9.20 Windows versions (as well as the 24.01 beta, of course).